### PR TITLE
Front load and cache some page chain class prep

### PR DESCRIPTION
**Pull request by gigabo**
This takes some work that used to happen for each request and caches it.

This is mostly groundwork for fine-grained instrumentation of page lifecycle methods.  It incidentally also slightly reduces the overhead of building up the page chain object during each request.

[`core/__bench__/handlePage.js`](https://github.com/redfin/react-server/pull/926) before:
```
1       0.000481308279135711
10      0.0009405891195623999
100     0.005067864883424908
1000    0.04754033069230767
```

[`core/__bench__/handlePage.js`](https://github.com/redfin/react-server/pull/926) after:
```
1       0.0004272630582241284
10      0.0006927164659505115
100     0.0034683026867557273
1000    0.031324801985714294
```

Basically reduces middleware overhead to from ~50µs to ~30µs.  Not a huge deal, but the instrumentation is likely to add a little overhead, so it's nice to pay some of that down ahead of time. :grin:

### CODE REVIEW

**Comment by gigabo** [comment]
Ugh, no idea why those zombie tests are failing.  Everything looks good browser-side in `react-server-test-pages` and `react-server-website`. :hurtrealbad:
